generator.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import argparse
+import subprocess
+import sys
+import asyncio
+
+DEFAULT_GIR_FILES_DIRECTORY = Path("./gir-files")
+DEFAULT_GIR_DIRECTORY = Path("./gir/")
+DEFAULT_GIR_PATH = DEFAULT_GIR_DIRECTORY / "target/release/gir"
+
+
+def run_command(command, folder=None):
+    return subprocess.run(command, cwd=folder, check=True)
+
+
+async def spawn_process(exe, args):
+    p = await asyncio.create_subprocess_exec(
+        str(exe),
+        *args,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+
+    stdout, stderr = await p.communicate()
+    assert p.returncode == 0, stderr.strip()
```

Comment:
What happens in case it panics? Also: `stderr.decode("utf-8").strip()`.

Reply:
Oh fair, I blindly moved this assert above the `stderr = stderr.decode("utf-8")`, that's why it was below at first...

> What happens in case it panics?

These asserts bubble up through the call stack and re-raise inside `asyncio.run`, when the task is awaited inside:

```python
        for coro in asyncio.as_completed(regen_crates(path, conf)):
            print(await coro)
```